experimental/Schemes/AffineSchemes.jl

```diff
@@ -543,10 +924,22 @@ function affine_space(kk::BRT, var_symbols::Vector{Symbol}) where {BRT<:Ring}
   return Spec(R)
 end
 
-@attr function dim(X::Spec)
+@attr function dim(X::AbsSpec{<:Ring, <:MPolyQuoLocalizedRing})
   return dim(saturated_ideal(localized_modulus(OO(X))))
 end
 
+@attr function dim(X::AbsSpec{<:Ring, <:MPolyLocalizedRing})
+  return ngens(ambient_ring(X))
+end
+
+@attr function dim(X::AbsSpec{<:Ring, <:MPolyRing})
```

Comment:
We already talked about this on the phone. Could you quickly document your suggestion here? What precisely is the oddity I have to take into account?

Reply:
Yes, sure. But I need to know what is the mathematical oddity that has to be addressed. I remember you said `dim(Spec(ZZ[x,y]) == 3`, because `dim(ZZ) == 1`. So what is the problem exactly and how do you suggest to resolve it? Should there be a function `dim(R::MPolyRing)` that we then use? Or should we do a manual case analysis of the `base_ring`?

Reply:
Yes, this is precisely the oddity: If  we are dealing with a polynomial ring over the integers in n variables, its dimension is n+1. If we are dealing with a quotient of a polynomial ring over the integers by some ideal, it makes a difference, whether the ideal contains an integer or not, but the backend Singular should take care of the oddity automatically as soon as it gets called.
 
OSCAR already has a function dim(Ideal) which returns the dimension of (Ring/Ideal), and we should use that applied to the ideal <0> in our ring here -- even if it feels odd. This shifts the case analysis to the commutative algebra, which in turn might or might not do it by itself or leave it to a backend like Singular -- in any case, the schemes-code are not the right place for the case analysis.